src/coreclr/vm/gchelpers.cpp

```diff
@@ -424,54 +530,12 @@ OBJECTREF AllocateSzArray(MethodTable* pArrayMT, INT32 cElements, GC_ALLOC_FLAGS
     }
     else
     {
-#ifndef FEATURE_64BIT_ALIGNMENT
-        if ((DATA_ALIGNMENT < sizeof(double)) && (pArrayMT->GetArrayElementType() == ELEMENT_TYPE_R8) &&
-            (totalSize < GCHeapUtilities::GetGCHeap()->GetLOHThreshold() - MIN_OBJECT_SIZE))
+        if ((DATA_ALIGNMENT < sizeof(double)) && (pArrayMT->GetArrayElementType() == ELEMENT_TYPE_R8))
         {
-            // Creation of an array of doubles, not in the large object heap.
-            // We want to align the doubles to 8 byte boundaries, but the GC gives us pointers aligned
-            // to 4 bytes only (on 32 bit platforms). To align, we ask for 12 bytes more to fill with a
-            // dummy object.
-            // If the GC gives us a 8 byte aligned address, we use it for the array and place the dummy
-            // object after the array, otherwise we put the dummy object first, shifting the base of
-            // the array to an 8 byte aligned address. Also, we need to make sure that the syncblock of the
-            // second object is zeroed. GC won't take care of zeroing it out with GC_ALLOC_ZEROING_OPTIONAL.
-            //
-            // Note: on 64 bit platforms, the GC always returns 8 byte aligned addresses, and we don't
-            // execute this code because DATA_ALIGNMENT < sizeof(double) is false.
-
-            _ASSERTE(DATA_ALIGNMENT == sizeof(double) / 2);
-            _ASSERTE((MIN_OBJECT_SIZE % sizeof(double)) == DATA_ALIGNMENT);   // used to change alignment
-            _ASSERTE(pArrayMT->GetComponentSize() == sizeof(double));
-            _ASSERTE(g_pObjectClass->GetBaseSize() == MIN_OBJECT_SIZE);
-            _ASSERTE(totalSize < totalSize + MIN_OBJECT_SIZE);
-            orArray = (ArrayBase*)Alloc(totalSize + MIN_OBJECT_SIZE, flags);
-
-            Object* orDummyObject;
-            if (((size_t)orArray % sizeof(double)) != 0)
-            {
-                orDummyObject = orArray;
-                orArray = (ArrayBase*)((size_t)orArray + MIN_OBJECT_SIZE);
-                if (flags & GC_ALLOC_ZEROING_OPTIONAL)
-                {
-                    // clean the syncblock of the aligned array.
-                    *(((void**)orArray)-1) = 0;
-                }
-            }
-            else
-            {
-                orDummyObject = (Object*)((size_t)orArray + totalSize);
-                if (flags & GC_ALLOC_ZEROING_OPTIONAL)
-                {
-                    // clean the syncblock of the dummy object.
-                    *(((void**)orDummyObject)-1) = 0;
-                }
-            }
-            _ASSERTE(((size_t)orArray % sizeof(double)) == 0);
-            orDummyObject->SetMethodTable(g_pObjectClass);
+            flags |= GC_ALLOC_ALIGN8;
+            orArray = (ArrayBase*)Alloc(totalSize, flags);
         }
         else
-#endif  // FEATURE_64BIT_ALIGNMENT
         {
 #ifdef FEATURE_64BIT_ALIGNMENT
```

Comment:
```suggestion
#ifdef FEATURE_DOUBLE_ALIGNMENT_HINT
            if (pArrayMT->GetArrayElementType() == ELEMENT_TYPE_R8)
            {
                flags |= GC_ALLOC_ALIGN8;
            }
#endif
#ifdef FEATURE_64BIT_ALIGNMENT
```
And delete the `if ((DATA_ALIGNMENT < sizeof(double)) && (pArrayMT->GetArrayElementType() == ELEMENT_TYPE_R8))` check above.

Reply:
So it means that there is only one code path without any check for FEATURE_64BIT_ALIGNMENT now?
``` 
else
{
#ifdef FEATURE_DOUBLE_ALIGNMENT_HINT
   if (pArrayMT->GetArrayElementType() == ELEMENT_TYPE_R8)
   {
      flags |= GC_ALLOC_ALIGN8;
   }
#endif
   orArray = (ArrayBase*)Alloc(totalSize, flags);
   orArray->SetMethodTable(pArrayMT);
}
```

Reply:
I meant to make it look like this:
```
#ifdef FEATURE_DOUBLE_ALIGNMENT_HINT
   if (pArrayMT->GetArrayElementType() == ELEMENT_TYPE_R8)
   {
      flags |= GC_ALLOC_ALIGN8;
   }
#endif
#ifdef FEATURE_64BIT_ALIGNMENT
    MethodTable* pElementMT = pArrayMT->GetArrayElementTypeHandle().GetMethodTable();
    if (pElementMT->RequiresAlign8() && pElementMT->IsValueType())
    {
        // This platform requires that certain fields are 8-byte aligned (and the runtime doesn't provide
        // this guarantee implicitly, e.g. on 32-bit platforms). Since it's the array payload, not the
        // header that requires alignment we need to be careful. However it just so happens that all the
        // cases we care about (single and multi-dim arrays of value types) have an even number of DWORDs
        // in their headers so the alignment requirements for the header and the payload are the same.
        _ASSERTE(((pArrayMT->GetBaseSize() - SIZEOF_OBJHEADER) & 7) == 0);
        flags |= GC_ALLOC_ALIGN8;
    }
#endif
   orArray = (ArrayBase*)Alloc(totalSize, flags);
   orArray->SetMethodTable(pArrayMT);
```

Reply:
Make more sense  :^)
✅ done